From 75e074717390fc80fc2167575464e727358462f3 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Wed, 2 Oct 2019 10:04:53 -0400
Subject: [PATCH] crypto: Make getrandom() non-blocking

getrandom() blocks by default.  In a xen stubdom, there is a lack of
entropy during boot, so this call hangs indefinitely.  The xen toolstack
doesn't see qemu identify itself as running and times out the stubdom
startup.

Switch to calling getrandom() as non-blocking during initialization
check.  In that case, an unitialized entropy pool with return eith
EAGAIN.  For qcrypto_random_init, that means we'll fall back to
/dev/urandom which warns to dmesg while still providing random data.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 crypto/random-platform.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/crypto/random-platform.c
+++ b/crypto/random-platform.c
@@ -45,7 +45,7 @@ int qcrypto_random_init(Error **errp)
     }
 #else
 # ifdef CONFIG_GETRANDOM
-    if (getrandom(NULL, 0, 0) == 0) {
+    if (getrandom(NULL, 0, GRND_NONBLOCK) == 0) {
         /* Use getrandom() */
         fd = -1;
         return 0;
